tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+   "metadata": {},
+   "source": [
+    "Since it is known that the BigEarthNet dataset from MLHub has a 1-to-1 pairing of source and labels, we can safely assume the first source item is the appropriate match for our label."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab77ef46-b54a-42e7-a780-bc8e8094ec6f",
+   "metadata": {},
+   "source": [
+    "This makes a connection to the Planetary Computer Data Catalog using the API endpoint URL."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38280716-e130-4de4-9699-2c0bcbfc056d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mspc_catalog = ps_client.open(MSPC_API_URL)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39f35d02-d2de-4be5-9317-c80214502c88",
+   "metadata": {},
+   "source": [
+    "We will now use the API client with the helper function above to fetch the best Landsat 8 match for the sampled label Item. This will find only the scenes where the label is completely within the scene, and there is minimal cloud cover."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53d3ab14-86c7-42a8-be96-8156f5e74d64",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "source_item, best_l8_match = get_landsat_8_match(first_label_item)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53556cf0-375b-4044-9262-d16de707a13d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "if best_l8_match:\n",
+    "    print(best_l8_match.id)\n",
+    "    print(best_l8_match.bbox)\n",
+    "    print(best_l8_match.properties)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f3b5d88-7f3c-4e74-8e68-2408ba762b0f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "explore_search_extent(ItemCollection([best_l8_match]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66c13807-50a3-4a8c-8a89-33002eafabe6",
+   "metadata": {},
+   "source": [
+    "If everything worked correctly, the geographic scope of the Landsat 8 scene should encompass a much larger surface area than the Sentinel-2 source and label chips. From here we need to crop the image down and make sure the chips from both products match."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "258a1d9e-55aa-4781-a0f5-e77ace273240",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_redirect_url(asset: Asset) -> str:\n",
+    "    \"\"\"Returns the direct URL to an asset.\n",
+    "\n",
+    "    Args:\n",
+    "        asset: Asset object from an Item\n",
+    "\n",
+    "    Returns:\n",
+    "        string response URL direct to Asset\n",
+    "    \"\"\"\n",
+    "    response = requests.get(asset.href, allow_redirects=True)\n",
+    "    if response.status_code == 200:\n",
+    "        return response.url\n",
+    "    return None"
```

Comment:
Do you ever hit this?

Reply:
Similar to my comments on `get_landsat_8_match` above, would you prefer that this be a more simplified notebook cell that stores the non-redirected Asset url in a variable, and not check the response status (e.g. leaving it to users and default error codes built into requests library)?